trustable_cli/metrics.py

```diff
@@ -35,6 +38,178 @@
+            self._update_message_size_metrics(event_data)
+
+    def get_commit_count(self):
+        return self.total_commits
+
+    def get_contributor_count(self):
+        return len(self.contributors)
+
+    def get_pony_factor(self):
+        """Number of individuals producing up to 50% of the total number of code contributions"""
+
+        partial_contributions = 0
+        pony_factor = 0
+
+        if len(self.contributors) == 0:
+            return 0
+
+        for _, contributions in self.contributors.most_common():
+            partial_contributions += contributions
+            pony_factor += 1
+            if partial_contributions / self.total_commits > 0.5:
+                break
+
+        return pony_factor
+
+    def get_elephant_factor(self):
+        """Number of companies producing up to 50% of the total number of code contributions"""
+
+        partial_contributions = 0
+        elephant_factor = 0
+
+        if len(self.companies) == 0:
+            return 0
+
+        for _, contributions in self.companies.most_common():
+            partial_contributions += contributions
+            elephant_factor += 1
+            if partial_contributions / self.total_commits > 0.5:
+                break
+
+        return elephant_factor
+
+    def get_file_type_metrics(self):
+        """Get the file type metrics"""
+
+        return dict(self.file_types)
+
+    def get_commit_size_metrics(self):
+        """Get the commit size metrics"""
+
+        metrics = {
+            "added_lines": self.added_lines,
+            "removed_lines": self.removed_lines,
+        }
+        return metrics
+
+    def get_message_size_metrics(self):
+        """Get the message size metrics"""
+
+        total = sum(self.messages_sizes)
+        number = len(self.messages_sizes)
+        average = 0
+        median = 0
+        if number > 0:
+            average = total / number
+            median = sorted(self.messages_sizes)[number // 2]
+
+        metrics = {
+            "total": total,
+            "average": average,
+            "median": median,
+        }
+        return metrics
+
+    def get_average_commits_week(self, days_interval: int):
+        """
+        Get the average number of commits per week
+
+        :param days_interval: Interval of days to calculate the average
+        """
+        return self.total_commits / days_interval / 7
+
+    def get_developer_categories(self):
+        """Return the number of core, regular and casual developers"""
+
+        core = 0
+        regular = 0
+        casual = 0
+        core_threshold = 0.3 * self.total_commits
+        regular_threshold = 0.1 * self.total_commits
+
+        for _, contributions in self.contributors.most_common():
+            if contributions >= core_threshold:
+                core += 1
+            elif contributions >= regular_threshold:
+                regular += 1
+            else:
+                casual += 1
```

Comment:
I think the next code is more similar to the implementation GrimoireLab has in cereslib. It has some special cases but for now we can use the same definition.

```suggestion
        regular_threshold = int(0.8 * self.total_commits)
        casual_threshold = int(0.95 * self.total_commits)
        acc_commits = 0
        
        for _, contributions in self.contributors.most_common():
            acc_commits += contributions
            
            if acc_commits <= core_threshold:
                core += 1
            elif contributions <= regular_threshold:
                regular += 1
            else:
                casual += 1
```